src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -179,25 +194,43 @@ public void transform(KeyValue<Schema, Map<String, String>> input,
    * @param config Salesforce Source Batch config
    * @return schema calculated from query
    */
-  private Schema getSchema(SalesforceSourceConfig config, OAuthInfo oAuthInfo) {
+  public static Schema getSchema(SalesforceSourceConfig config, OAuthInfo oAuthInfo) {
     String query = config.getQuery(System.currentTimeMillis(), oAuthInfo);
     SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromQuery(query);
     try {
       AuthenticatorCredentials credentials = new AuthenticatorCredentials(oAuthInfo,
-                                                                          config.getConnection().getConnectTimeout(),
-                                                                          config.getConnection().getProxyUrl());
+          config.getConnection().getConnectTimeout(),
+          config.getConnection().getProxyUrl());
       return SalesforceSchemaUtil.getSchema(credentials, sObjectDescriptor);
     } catch (ConnectionException e) {
       String errorMessage = SalesforceConnectionUtil.getSalesforceErrorMessageFromException(e);
       throw new RuntimeException(
-        String.format("Failed to get schema from the query '%s': %s", query, errorMessage),
-        e);
+          String.format("Failed to get schema from the query '%s': %s", query, errorMessage),
+          e);
+    }
+  }
+
+  /**
+   * Get the RecordReader for a given split.
+   *
+   * @param salesforceSplit Given Salesforce Split
+   * @param schemaJsonString Schema JSON
+   * @param credentials Credential object
+   * @return RecordReader for the given split
+   */
+  public static RecordReader<Schema, Map<String, ?>> getRecordReader (SalesforceSplit salesforceSplit,
+                                                                      String schemaJsonString,
+                                                                      AuthenticatorCredentials credentials) {
```

Comment:
Why do we need this method in this class? Can caller just call the createRecordReader directly?

Reply:
I believe we can.